### ISSUE TITLE
Make library compatible with PHP 7

### ISSUE THREAD

**Issue by henriquemoody** [Respect/Validation#362]
Some reserved words were introduced on PHP 7:

```
Cannot use 'Bool' as class name as it is reserved in library/Rules/Bool.php on line 14
Cannot use 'False' as class name as it is reserved in library/Rules/False.php on line 14
Cannot use 'Float' as class name as it is reserved in library/Rules/Float.php on line 14
Cannot use 'Int' as class name as it is reserved in library/Rules/Int.php on line 14
Cannot use 'String' as class name as it is reserved in library/Rules/String.php on line 14
Cannot use 'True' as class name as it is reserved in library/Rules/True.php on line 14
```

**Comment by oschwald** [comment]
:+1:

**Comment by henriquemoody** [comment]
Does anyone have names' suggestion?

I suggest these changes based on `nullValue()` and some other things that make sense to me:

- Rename `bool()` to `boolType()` - Also split it to `boolValue()`
- Rename `false()` to `falseValue()` - Also split it to `falsityValue()`
- Rename `float()` to `floatType()` - Also split it to `floatValue()`
- Rename `int()` to `intType()` - Also split it to `intValue()`
- Rename `string()` to `stringType()`
- Rename `true()` to `trueValue()`- Also split it to `truthValue()`

In this context, we should also:
- Create `callableType()`
- Create `resourceType()`
- Create `scalarValue()`
- Rename `arr()` to `arrayType()`
- Rename `numeric()` to `numericValue()`
- Rename `object()` to `objectType()`


***
Any thoughts? @Respect/troubleshooters?

**Comment by wesleyvicthor** [comment]
callableType :+1:

**Comment by henriquemoody** [comment]
Fixed by 349a45d, 21c8f16, 0e3d1e7, 92f68b6, b297f6e, 92f0eb5.
